beamer/agent/chain.py

```diff
@@ -591,18 +591,22 @@ def _timestamp_is_l1_finalized(
 
 
 def maybe_prove(claim: Claim, context: Context) -> None:
-    # mainnet: 10, goerli: 420, local: 901
-    if context.target_chain.id not in (10, 420, 901):
-        claim.message_proved = True
-        return
-
-    if claim.message_proved:
-        return
-
     request = context.requests.get(claim.request_id)
 
     assert request is not None, "Active claim for non-existent request"
 
+    proved_tx = None
+    if request.fill_tx is not None:
+        proved_tx = request.fill_tx
+    elif claim.invalidation_tx is not None:
+        proved_tx = claim.invalidation_tx
+
+    # mainnet: 10, goerli: 420, local: 901
+    if context.target_chain.id not in (10, 420, 901):
+        if claim.proved_tx is None:
```

Comment:
This is not correct. In the case of Non-OP chains the claim.proved_tx must be set unconditionally. (fill_tx should overwrite invalidation_tx)